spec/ddtrace/configuration/base_spec.rb

```diff
@@ -39,7 +39,7 @@
     describe 'instance behavior' do
       subject(:base_object) { base_class.new }
 
-      it { is_expected.to be_a_kind_of(Datadog::Environment::Helpers) }
+      it { is_expected.to be_a_kind_of(Datadog::Core::Environment::VariableHelpers) }
```

Comment:
I understand why we're testing this, but I think RSpec here is really forcing us to write down something that I think underlines that our current approach to module usage is wrong: 

`Datadog::Configuration::Base instance behavior is expected to be a kind of VariableHelpers`.

Really? Configuration::Base is a kind of VariableHelpers? I totally thought it was a tool for getting settings 😉

Reply:
I'm guessing a better option here would be to directly reference `Environment.env_to_bool`, instead of mixing it in?

We could mix it in as private methods?

I'm not sure what the ideal solution is here.

Reply:
I don't think changing this is in scope for this PR, but I think there's two things at play here:

* I argue that this test is a bit too coupled to the implementation. There's multiple ways in which these helpers could be made usable to the base class, and the test is kinda forcing one.
* For the implementation, I think a `Forwardable` would be a nice option.